backend/src/main/java/com/ziwg/reservatio/controllers/EmployeeController.java

```diff
@@ -32,41 +31,74 @@ public EmployeeController(EmployeeRepository employeeRepository,
     }
 
-    @DeleteMapping("deleteEmployee/{employeeId}")
-    public ResponseEntity<String> deleteEmployee(@PathVariable Long employeeId) {
-        Optional<Employee> employeeToDelete = employeeRepository.findById(employeeId);
+    private Employee createEmployee(EmployeePojo employeePojo, ServiceProvider serviceProvider) {
+        return Employee.builder().firstName(employeePojo.getFirstName()).lastName(employeePojo.getLastName()).serviceProvider(serviceProvider).build();
+    }
+
+    private void setServicesForEmployee(EmployeePojo employeePojo, Employee employee) {
+        List<Service> servicesToAdd = getServices(employeePojo, employee.getServiceProvider());
+        employee.setServices(servicesToAdd);
+    }
+
+    private List<Service> getServices(EmployeePojo employeePojo, ServiceProvider serviceProvider) {
+        return serviceProvider.getServices().stream().filter(service -> employeePojo.getServices().contains(service.getName())).collect(Collectors.toList());
+    }
+
+    @DeleteMapping("{employeeId}")
+    public ResponseEntity<String> deleteEmployee(@PathVariable Long serviceProviderId, @PathVariable Long employeeId) {
+        Optional<ServiceProvider> serviceProvider = serviceProviderRepository.findById(serviceProviderId);
+        if (serviceProvider.isEmpty())
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        Optional<Employee> employeeToDelete = getEmployeeFromServiceProvider(employeeId, serviceProvider.get());
         if (employeeToDelete.isEmpty())
             return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
-        if (employeeToDelete.get().getReservations().stream().anyMatch(reservation -> reservation.getDateTime().isAfter(LocalDateTime.now())))
+        if (hasAnyUpcomingReservations(employeeToDelete.get()))
             return new ResponseEntity<>("Cannot delete an employee that has upcoming appointments", HttpStatus.BAD_REQUEST);
-        ServiceProvider serviceProvider = employeeToDelete.get().getServiceProvider();
+        deleteEmployeeFromServiceProvider(employeeId, serviceProvider.get());
+        clearRelationsInEmployee(employeeToDelete.get());
+        return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+    }
+
+    private Optional<Employee> getEmployeeFromServiceProvider(Long employeeId, ServiceProvider serviceProvider) {
+        return serviceProvider.getEmployees().stream().filter(employee -> employee.getId().equals(employeeId)).findAny();
+    }
+
+    private boolean hasAnyUpcomingReservations(Employee employeeToDelete) {
+        return employeeToDelete.getReservations().stream().anyMatch(reservation -> reservation.getDateTime().isAfter(LocalDateTime.now()));
+    }
+
+    private void deleteEmployeeFromServiceProvider(Long employeeId, ServiceProvider serviceProvider) {
         serviceProvider.setEmployees(serviceProvider.getEmployees().stream().filter(employee -> !employee.getId().equals(employeeId)).collect(Collectors.toList()));
```

Comment:
To co jest w argumencie tej metody powinno być raczej osobną zmienną, łatwiej będzie przeczytać co tu się dzieje.